ethereum/referencetests/src/main/java/org/hyperledger/besu/ethereum/referencetests/ReferenceTestProtocolSchedules.java

```diff
@@ -42,48 +42,46 @@ public class ReferenceTestProtocolSchedules {
     builder.put(
         "EIP158ToByzantiumAt5",
-        createSchedule(new StubGenesisConfigOptions().eip158Block(0).byzantiumBlock(5)));
-    builder.put("Byzantium", createSchedule(new StubGenesisConfigOptions().byzantiumBlock(0)));
-    builder.put(
-        "Constantinople", createSchedule(new StubGenesisConfigOptions().constantinopleBlock(0)));
-    builder.put(
-        "ConstantinopleFix", createSchedule(new StubGenesisConfigOptions().petersburgBlock(0)));
-    builder.put("Petersburg", createSchedule(new StubGenesisConfigOptions().petersburgBlock(0)));
-    builder.put("Istanbul", createSchedule(new StubGenesisConfigOptions().istanbulBlock(0)));
-    builder.put("MuirGlacier", createSchedule(new StubGenesisConfigOptions().muirGlacierBlock(0)));
-    builder.put("Berlin", createSchedule(new StubGenesisConfigOptions().berlinBlock(0)));
-    builder.put(
-        "London",
-        createSchedule(new StubGenesisConfigOptions().londonBlock(0).baseFeePerGas(0x0a)));
-    builder.put(
-        "ArrowGlacier", createSchedule(new StubGenesisConfigOptions().arrowGlacierBlock(0)));
-    builder.put("GrayGlacier", createSchedule(new StubGenesisConfigOptions().grayGlacierBlock(0)));
-    builder.put(
-        "Merge",
-        createSchedule(new StubGenesisConfigOptions().mergeNetSplitBlock(0).baseFeePerGas(0x0a)));
-    builder.put(
-        "Shanghai",
-        createTimestampSchedule(
-            new StubGenesisConfigOptions().shanghaiTime(0).baseFeePerGas(0x0a)));
-    builder.put(
-        "Cancun",
-        createTimestampSchedule(new StubGenesisConfigOptions().cancunTime(0).baseFeePerGas(0x0a)));
+        createSchedule(genesisStub.clone().eip158Block(0).byzantiumBlock(5)));
+    builder.put("Byzantium", createSchedule(genesisStub.clone().byzantiumBlock(0)));
+    builder.put("Constantinople", createSchedule(genesisStub.clone().constantinopleBlock(0)));
+    builder.put("ConstantinopleFix", createSchedule(genesisStub.clone().petersburgBlock(0)));
+    builder.put("Petersburg", createSchedule(genesisStub.clone().petersburgBlock(0)));
+    builder.put("Istanbul", createSchedule(genesisStub.clone().istanbulBlock(0)));
+    builder.put("MuirGlacier", createSchedule(genesisStub.clone().muirGlacierBlock(0)));
+    builder.put("Berlin", createSchedule(genesisStub.clone().berlinBlock(0)));
+
+    // the following schedules activate EIP-1559, but may have non-default
+    if (genesisStub.getBaseFeePerGas().isEmpty()) {
+      genesisStub.baseFeePerGas(0x0a);
+    }
+    builder.put("London", createSchedule(genesisStub.clone().londonBlock(0)));
+    builder.put("ArrowGlacier", createSchedule(genesisStub.clone().arrowGlacierBlock(0)));
+    builder.put("GrayGlacier", createSchedule(genesisStub.clone().grayGlacierBlock(0)));
+    builder.put("Merge", createSchedule(genesisStub.clone().mergeNetSplitBlock(0)));
+    builder.put("Merge+3545+3670", createTimestampSchedule(genesisStub.clone().shanghaiTime(0)));
```

Comment:
I guess these are EVM EIPS - I can't find 3545 though?

Reply:
These specific values were extracted from a particular set of `develop` branch tests from the reference test.  Since we cannot easily handle "fork+EIP" they are encoded here to provide testing for EOFv1 tests until they are formally sunset.